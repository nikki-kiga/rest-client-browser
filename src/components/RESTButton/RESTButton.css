input[type=radio] {
    visibility: hidden;
  height: 0;
  width: 0;
}

input[type='radio']:checked + label{
    border: 1px solid #ccc;
    border-radius: 5px;
    cursor: pointer;
    margin-right: .5rem;
    padding: 0 1rem;
    line-height: 2rem;
    height: 2rem;
    border-bottom: 2px solid blue;
}

label {
    background: #f5f5f5;
    border: 1px solid #ccc;
    border-radius: 5px;
    cursor: pointer;
    margin-right: .5rem;
    padding: 0 1rem;
    line-height: 2rem;
    height: 2rem;
}
