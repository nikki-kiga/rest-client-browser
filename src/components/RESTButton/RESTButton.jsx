import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ reqType, onChange  }) => (
  <div className = 'radio'>
    <label>
      <input 
        type= 'radio'
        name= 'requestType'
        value={reqType}
        onChange={onChange}
        className='request-type'
      />
      {reqType.toUpperCase()}
    </label>
  </div>
);

Button.propTypes = {
  reqType: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

export default Button;
