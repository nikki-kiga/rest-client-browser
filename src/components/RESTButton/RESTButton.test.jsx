import React from 'react';
import { shallow } from 'enzyme';
import RESTButton from './RESTButton.jsx';

describe('REST Button', () => {
  let wrapper;
  let onChange;
  beforeEach(() => {
    onChange = jest.fn();
    wrapper = shallow(<RESTButton reqType='test' onChange={onChange}/>);
  });
  it('renders a button based on the input', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has a radio input', () => {
    expect(wrapper.find('input[name="requestType"]').prop('value')).toEqual('test');
  });
});
