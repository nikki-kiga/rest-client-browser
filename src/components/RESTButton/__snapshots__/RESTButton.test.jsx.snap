// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`REST Button renders a button based on the input 1`] = `
<div
  className="radio"
>
  <label>
    <input
      className="request-type"
      name="requestType"
      onChange={[MockFunction]}
      type="radio"
      value="test"
    />
    TEST
  </label>
</div>
`;
